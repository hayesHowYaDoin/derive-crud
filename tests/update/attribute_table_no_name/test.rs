use derive_crud::Update;

#[derive(Update)]
#[crud_table]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let _result = Test {
        id: 1,
        name: "Debbie".to_string(),
    }
    .update()
    .await;
}
