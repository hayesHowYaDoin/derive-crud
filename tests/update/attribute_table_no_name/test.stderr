error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/update/attribute_table_no_name/test.rs:4:1
  |
4 | / #[crud_table]
5 | | struct Test {
6 | |     #[crud_id]
7 | |     id: i64,
8 | |     name: String,
9 | | }
  | |_^

error[E0599]: no method named `update` found for struct `Test` in the current scope
  --> tests/update/attribute_table_no_name/test.rs:20:6
   |
5  |   struct Test {
   |   ----------- method `update` not found for this struct
...
16 |       let _result = Test {
   |  ___________________-
17 | |         id: 1,
18 | |         name: "Debbie".to_string(),
19 | |     }
20 | |     .update(&pool)
   | |     -^^^^^^ method not found in `Test`
   | |_____|
   |
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `update`, perhaps you need to implement one of them:
           candidate #1: `digest::Update`
           candidate #2: `digest::digest::Digest`
           candidate #3: `digest::digest::DynDigest`
